src/operator/csv.rs

```diff
@@ -19,13 +22,14 @@ use std::{borrow::Cow, io::Read, marker::PhantomData};
 /// A source operator that reads records of type `T` from a CSV file.
 ///
 /// The operator reads the entire file and yields its contents
-/// in the first clock cycle.
-pub struct CsvSource<R, C, T> {
+/// in the first clock cycle as a Z-set with unit weights.
+pub struct CsvSource<R, T, W, C> {
```

Comment:
I am wondering whether it isn't better to have this actually return tuples and use the makezset operator afterwards. This is one weakness of DD, that everything is a zset, but it doesn't have to be in DBSP.
If that's inefficient you can have separate methods to yield tuples and zsets.